### ISSUE TITLE
Failed to install the latest version of the migration script.

### ISSUE THREAD

**Issue by rbirkgit** [nrwl/nx#23222]
### Current Behavior

I keep getting this error every time I update Nx. Is this normal or am I doing something wrong?

`Failed to install the latest version of the migration script. Using the current version.`

Then it says there are no new migrations and only updates Nx versions.

### Expected Behavior

No errors.

### Steps to Reproduce

1. nx migrate latest

### Nx Report

```shell
Node   : 18.19.0
OS     : linux-x64
yarn   : 4.0.1

nx                 : 18.3.4
@nx/js             : 18.3.4
@nx/linter         : 18.3.4
@nx/eslint         : 18.3.4
@nx/workspace      : 18.3.4
@nx/angular        : 18.3.4
@nx/devkit         : 18.3.4
@nx/eslint-plugin  : 18.3.4
@nrwl/tao          : 18.3.4
@nx/web            : 18.3.4
@nx/webpack        : 18.3.4
typescript         : 5.4.5
---------------------------------------
Community plugins:
ng-mocks : 14.12.2
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by leosvelperez** [comment]
Thanks for reporting this!

Could you please run with `--verbose` or `NX_VERBOSE_LOGGING=true` and share the error logs? Are you trying to migrate from Nx 18.3.4?

**Comment by rbirkgit** [comment]
Yes, this time from 18.3.4 to 19.0.0. Got same issues with previous updates.

Let me try with added verbose info.

**Comment by rbirkgit** [comment]
I ran this just now (18.3.4->19.0.1). Same error. --verbose didn't output anything more or any error logs.

```
root@RBIRK:/mnt/d/Git/cloud-ui# npx nx migrate latest --verbose
Failed to install the latest version of the migration script. Using the current version.
```

**Comment by leosvelperez** [comment]
I see. Could you try running `NX_VERBOSE_LOGGING=true npx nx migrate latest`?

**Comment by rbirkgit** [comment]
Got this now:
```
<ref *1> Error: spawnSync /bin/sh ENOENT
    at Object.spawnSync (node:internal/child_process:1117:20)
    at spawnSync (node:child_process:876:24)
    at execSync (node:child_process:957:15)
    at nxCliPath (/mnt/d/Git/cloud-automation-ui/node_modules/nx/src/command-line/migrate/command-object.js:131:46)
    at runMigration (/mnt/d/Git/cloud-automation-ui/node_modules/nx/src/command-line/migrate/command-object.js:90:19)
    at Object.handler (/mnt/d/Git/cloud-automation-ui/node_modules/nx/src/command-line/migrate/command-object.js:18:9)
    at /mnt/d/Git/cloud-automation-ui/node_modules/yargs/build/index.cjs:1:8993
    at j (/mnt/d/Git/cloud-automation-ui/node_modules/yargs/build/index.cjs:1:4956)
    at _.handleValidationAndGetResult (/mnt/d/Git/cloud-automation-ui/node_modules/yargs/build/index.cjs:1:8962)
    at _.applyMiddlewareAndGetResult (/mnt/d/Git/cloud-automation-ui/node_modules/yargs/build/index.cjs:1:9604) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawnSync /bin/sh',
  path: '/bin/sh',
  spawnargs: [ '-c', 'yarn set version 4.0.1' ],
  error: [Circular *1],
  status: null,
  signal: null,
  output: null,
  pid: 0,
  stdout: null,
  stderr: null
}
```

**Comment by rbirkgit** [comment]
I tried directly in windows terminal and got same error. In the past migration has never worked well directly in windows, and why I switched to WSL.

Just in case, these are the difference from error above on windows. The rest of the error is identical.
```
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawnSync C:\\WINDOWS\\system32\\cmd.exe',
  path: 'C:\\WINDOWS\\system32\\cmd.exe',
  spawnargs: [ '/d', '/s', '/c', '"yarn set version 4.0.1"' ],
  ```

**Comment by leosvelperez** [comment]
Thanks for the info! I've identified the issue and I'll push a fix.

**Comment by rbirkgit** [comment]
I still get the same error updating to 19.0.2.

**Comment by rbirkgit** [comment]
@leosvelperez getting same error in my WSL ubuntu env too with 19.0.2

**Comment by leosvelperez** [comment]
@rbirkgit it will still happen until you're migrated to Nx 19.0.2. After that, the error will go away. The reason is that the version you have installed has the issue. Only when you have Nx 19.0.2 (the version with the fix), the issue will go away.

**Comment by rbirkgit** [comment]
That explain it! Thanks!

**Comment by nhhockeyplayer** [comment]
I have 19.2.2 and its showing up
just an FYI

```
% nx report
 NX  Falling back to ts-node for local typescript execution. This may be a little slower.
  - To fix this, ensure @swc-node/register and @swc/core have been installed

 NX   Report complete - copy this into the issue template

Node   : 22.2.0
OS     : darwin-x64
yarn   : 1.22.22

nx (global)        : 19.2.2
nx                 : 19.2.2
@nx/js             : 19.2.2
@nx/jest           : 19.2.2
@nx/linter         : 19.2.2
@nx/eslint         : 19.2.2
@nx/workspace      : 19.2.2
@nrwl/workspace    : 8.12.11
@nx/angular        : 19.2.2
@nx/cypress        : 19.2.2
@nx/devkit         : 19.2.2
@nrwl/devkit       : 13.10.6
@nx/eslint-plugin  : 19.2.2
@nx/express        : 19.2.2
@nx/nest           : 19.2.2
@nx/node           : 19.2.2
@nx/storybook      : 19.2.2
@nrwl/tao          : 19.2.2
@nx/web            : 19.2.2
@nx/webpack        : 19.2.2
@nrwl/schematics   : 8.12.11
typescript         : 5.4.5
---------------------------------------
Community plugins:
@compodoc/compodoc                    : 1.1.25
@fortawesome/angular-fontawesome      : 0.11.1
@ionic/angular                        : 6.7.5
@ngneat/transloco                     : 4.3.0
@ngrx/component                       : 17.0.1
@ngrx/component-store                 : 17.0.1
@ngrx/effects                         : 17.0.1
@ngrx/entity                          : 17.0.1
@ngrx/router-store                    : 17.0.1
@ngrx/schematics                      : 17.0.1
@ngrx/store                           : 17.0.1
@ngrx/store-devtools                  : 17.0.1
@nxtend/capacitor                     : 13.0.0
@nxtend/ionic-angular                 : 13.1.0
@storybook/angular                    : 7.6.6
angular-builder-custom-terser-options : 1.0.1
apollo-angular                        : 4.2.1
---------------------------------------
The following packages should match the installed version of nx
  - @nrwl/workspace@8.12.11
  - @nrwl/devkit@13.10.6

To fix this, run `nx migrate nx@19.2.2`

```
